bgpd/bgp_vty.c

```diff
@@ -10471,6 +10471,34 @@ static void bgp_show_failed_summary(struct vty *vty, struct bgp *bgp,
 	}
 }
 
+/* If the peer's description includes whitespaces
+ * then return the first occurrence. Also strip description
+ * to the given size if needed.
+ */
+static char *bgp_peer_description_stripped(char *desc, uint8_t size)
+{
+	char **desc_occurrences;
+	char *desc_stripped;
+	char buf[BUFSIZ] = {};
+	int num_occurrences;
+
+	frrstr_split(desc, " ", &desc_occurrences, &num_occurrences);
+
+	if (num_occurrences > 1)
+		snprintf(buf, size ? size : sizeof(buf), "%s",
+			 desc_occurrences[0]);
+	else
+		snprintf(buf, size ? size : sizeof(buf), "%s", desc);
+
+	desc_stripped = buf;
+
+	for (int j = 0; j < num_occurrences; j++)
+		XFREE(MTYPE_TMP, desc_occurrences[j]);
+
+	XFREE(MTYPE_TMP, desc_occurrences);
+
```

Comment:
is it possible to strip without having to allocate buffers ?

Reply:
I'll do.

Reply:
Updated.